tests/common/beamlines/test_device_factory.py

```diff
@@ -0,0 +1,90 @@
+from bluesky.run_engine import RunEngine
+from ophyd_async.epics.motion.motor import Motor
+from ophyd_async.plan_stubs.ensure_connected import ensure_connected
+
+from dodal.common.beamlines.beamline_utils import ACTIVE_DEVICES
+from dodal.common.beamlines.device_factory import (
+    DeviceInitializationConfig,
+    device_factory,
+)
+
+
+def test_terminal_use_case_decorated_motor_not_mock(RE: RunEngine):
+    @device_factory()
+    def motor(name: str = "motor", prefix: str = "motor:"):
+        return Motor(name=name, prefix=prefix)
+
+    m = motor(name="foo", mock=True)
+
+    assert m is not None
+    assert m.name == "foo"
+
+    RE(ensure_connected(m))
+
+
+def test_terminal_use_case_decorated_motor_mock(RE: RunEngine):
+    @device_factory(mock=True)
+    def motor(name: str = "motor", prefix: str = "motor:"):
+        return Motor(name=name, prefix=prefix)
+
+    m = motor()
+    RE(ensure_connected(m))
+
+    assert m is not None
+    assert m.name == "motor"
+
+
+def test_decorator_directly_with_name_override(RE: RunEngine):
+    ACTIVE_DEVICES.clear()
+    beamline_prefix = "example:"
+
+    @device_factory(mock=True, use_factory_name=False)
+    def m2():
+        return Motor(name="foo", prefix=f"{beamline_prefix}xyz:")
+
+    device = m2()
+    assert device is not None
+    assert "foo" in ACTIVE_DEVICES.keys()
```

Comment:
```suggestion
    ACTIVE_DEVICES.clear()

    @device_factory(mock=True, use_factory_name=False)
    def m2():
        return Motor(name="foo", prefix="xyz:")

    device = m2()
    assert device is not None
    assert ACTIVE_DEVICES["foo"] is device
```

the f-string part doesn't do anything and this more explicitly tests the value of the cached device